src/app/portlets/dot-edit-content/dot-edit-content.component.spec.ts

```diff
@@ -91,4 +93,28 @@ describe('DotEditContentComponent', () => {
         expect(loadingIndicatorElem).not.toBeNull();
         expect(spyLoadingIndicator).toHaveBeenCalled();
     });
+
+    it('should display confirmation dialog and remove contentlet when user accepts', () => {
+        component.ngOnInit();
+        const mockResEvent = {
+            contentletEvents: {},
+            dataset: {
+                dotIdentifier: '2sfasfk-sd2d-4dxc-sdfnsdkjnajd0',
+                dotInode: '26ad1jbj-23xd-4cx3-9cf2-432scc413cc2'
+            },
+            event: 'remove'
+        };
+        const dotEditContentHtmlService = fixture.debugElement.injector.get(DotEditContentHtmlService);
+
+        spyOn(dotEditContentHtmlService, 'contentletEvents').and.returnValue(Observable.of(mockResEvent));
+        spyOn(dotEditContentHtmlService, 'removeContentlet').and.callFake(res => {});
```

Comment:
we should not make a mock of 'removeContentlet' because this is private method of dot-edit-content.component.ts, if this method's implementation is wrong the test keep passing.
In this case we should mokc just the 'this.dotConfirmationService.confirm' method.

Reply:
I think you are confusing the private method of the component called 'removeContentlet' with the service method "removeContentlet", So I was testing dotEditContentHtmlService.removeContentlet was called with the mock data